doc/radosgw/s3select.rst

```diff
@@ -29,31 +29,38 @@ Basic workflow
     | The current implementation supports CSV objects and since chunks are randomly “cutting” the CSV rows in the middle, those broken-lines (first or last per chunk) are skipped while processing the query.   
     | Those “broken” lines are stored and later merged with the next broken-line (belong to the next chunk), and finally processed.
    
     | Per each processed chunk an output message is formatted according to `AWS specification <https://docs.aws.amazon.com/AmazonS3/latest/API/archive-RESTObjectSELECTContent.html#archive-RESTObjectSELECTContent-responses>`_ and sent back to the client.
     | RGW supports the following response: ``{:event-type,records} {:content-type,application/octet-stream} {:message-type,event}``.
-    | For aggregation queries the last chunk should be identified as the end of input, following that the s3-select-engine initiates end-of-process and produces an aggregate result.  
+    | For aggregation queries the last chunk should be identified as the end of input, following that the s3-select-engine initiates end-of-process and produces an aggregated result.  
 
         
 Basic functionalities
 ~~~~~~~~~~~~~~~~~~~~~
 
-    | **S3select** has a definite set of functionalities that should be implemented (if we wish to stay compliant with AWS), currently only a portion of it is implemented.
+    | **S3select** has a definite set of functionalities compliant with AWS.
     
-    | The implemented software architecture supports basic arithmetic expressions, logical and compare expressions, including nested function calls and casting operators, that alone enables the user reasonable flexibility. 
+    | The implemented software architecture supports basic arithmetic expressions, logical and compare expressions, including nested function calls and casting operators, which enables the user great flexibility. 
     | review the below s3-select-feature-table_.
 
 
 Error Handling
 ~~~~~~~~~~~~~~
 
-    | Any error occurs while the input query processing, i.e. parsing phase or execution phase, is returned to client as response error message.
-
-    | Fatal severity (attached to the exception) will end query execution immediately, other error severity are counted, upon reaching 100, it ends query execution with an error message.
-
+    | Upon an error being detected, a specific error message sends back to the client.
+    | Currently, there are 2 main types of error.
+    |
+    | **Syntax error**: the s3selecet parser rejects user requests that are not aligned with parser syntax definitions, as     
+    | described in this documentation.
+    | Upon Syntax Error, the engine creates an error message that points to the location of the error.
+    | RGW sends back the error message in a specific error response. 
+    |
+    | **Processing TIme error**: the runtime engine may detect errors that occur only on processing time, for that type of     
```

Comment:
the whole execution is on runtime.
Processing time, in this context, means that, 
query syntax is OK, the data processing(the object itself) had begun,  
at some point, the query execution encounters an error (string instead of a number), thus it needs different error reporting.

syntax-error reject the s3-operation immediately, while processing-time may occur much later.